homeassistant/components/weatherflow/sensor.py

```diff
@@ -0,0 +1,378 @@
+        suggested_display_precision=1,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="dew_point_temperature",
+        translation_key="dew_point",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="feels_like_temperature",
+        translation_key="feels_like",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="wet_bulb_temperature",
+        translation_key="wet_bulb_temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="battery",
+        translation_key="battery_voltage",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="illuminance",
+        native_unit_of_measurement=LIGHT_LUX,
+        device_class=SensorDeviceClass.ILLUMINANCE,
+        state_class=SensorStateClass.MEASUREMENT,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="lightning_strike_average_distance",
+        icon="mdi:lightning-bolt",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DISTANCE,
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        translation_key="lightning_average_distance",
+        suggested_display_precision=2,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="lightning_strike_count",
+        translation_key="lightning_count",
+        icon="mdi:lightning-bolt",
+        state_class=SensorStateClass.TOTAL,
+        raw_data_conv_fn=lambda raw_data: raw_data,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="precipitation_type",
+        translation_key="precipitation_type",
+        device_class=SensorDeviceClass.ENUM,
+        options=["none", "rain", "hail", "rain_hail", "unknown"],
```

Comment:
If the value is unknown, can we return None? To make this raw_data_conv_fn cleaner I would suggest putting it in a separate function.

When the value is None, HA already shows it as Unknown

Reply:
updated.. :)